In [1]:
import pandas as pd

data=pd.read_csv("fusionned_file.csv",sep=";")
data.head()

source sentence  \
0   :[0],"non-coding RNAs (lncRNAs) may play crit...   
1  Recently, lncRNAs have been widely identified ...   
2  However, there is rare information about them ...   
3  During humanitarian crises, there is a need fo...   
4  Such need creates the base for misinformation ...   

                                 simplified sentence                   run_id  \
0   Long non-coding RNAs, or lncRNAs, may play si...  UBO_Task3.1_Phi4mini-sl   
1   Small RNA molecules are found in complex cell...  UBO_Task3.1_Phi4mini-sl   
2   There is unusual information here that doesn'...  UBO_Task3.1_Phi4mini-sl   
3  When crises happen, people need quick and larg...  UBO_Task3.1_Phi4mini-sl   
4  This reduction helps in combating climate chan...  UBO_Task3.1_Phi4mini-sl   

               snt_id No error  Random generation  Syntax error  \
0  G04.2_2519200480_1     VRAI              False         False   
1  G04.2_2519200480_2     VRAI              False         False   
2  G04.2_2519200480_3     FAUX              False         False   
3  G07.1_3007380484_1     FAUX              False         False   
4  G07.1_3007380484_2     FAUX              False         False   

   Contradiction  Simple punctuation / grammar errors  Redundancy  \
0          False                                False       False   
1          False                                False       False   
2          False                                False       False   
3          False                                False       False   
4          False                                False        True   

   Format misalignement  Prompt misalignement  Out-of-Scope Generation  \
0                 False                 False                    False   
1                 False                 False                    False   
2                 False                 False                     True   
3                 False                 False                     True   
4                 False                  True                    False   

   Topic shift  Oversimplification of Logical Arguments  Overgeneralization  \
0        False                                    False               False   
1        False                                    False               False   
2         True                                    False               False   
3        False                                    False               False   
4         True                                    False               False   

   Loss of Informative Content  Factuality hallucination  \
0                        False                     False   
1                        False                     False   
2                        False                     False   
3                        False                     False   
4                        False                     False   

   Faithfulness hallucination Commentaire  
0                       False         NaN  
1                       False         NaN  
2                       False         NaN  
3                       False         NaN  
4                       False         NaN

In [2]:
data.describe()

source sentence  \
count                                                 330   
unique                                                330   
top      :[0],"non-coding RNAs (lncRNAs) may play crit...   
freq                                                    1   

                                      simplified sentence  \
count                                                 330   
unique                                                329   
top     tries desire desire confirms confirms confirms...   
freq                                                    2   

                                       run_id              snt_id No error  \
count                                     330                 330      330   
unique                                     29                 330        2   
top     UZH_Pandas_Task3.1_selection_with_sle  G04.2_2519200480_1     FAUX   
freq                                       30                   1      184   

       Random generation Syntax error Contradiction  \
count                330          330           330   
unique                 2            2             1   
top                False        False         False   
freq                 315          297           330   

       Simple punctuation / grammar errors Redundancy Format misalignement  \
count                                  330        330                  330   
unique                                   2          2                    2   
top                                  False      False                False   
freq                                   300        306                  313   

       Prompt misalignement Out-of-Scope Generation Topic shift  \
count                   330                     330         330   
unique                    2                       2           2   
top                   False                   False       False   
freq                    326                     284         311   

       Oversimplification of Logical Arguments Overgeneralization  \
count                                      330                330   
unique                                       2                  2   
top                                      False              False   
freq                                       305                288   

       Loss of Informative Content Factuality hallucination  \
count                          330                      330   
unique                           2                        2   
top                          False                    False   
freq                           262                      327   

       Faithfulness hallucination  \
count                         330   
unique                          2   
top                         False   
freq                          303   

                                              Commentaire  
count                                                  33  
unique                                                 31  
top     pour le 'Topic Shift', j'ai considéré que le f...  
freq                                                    2

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 330 entries, 0 to 329
Data columns (total 20 columns):
 #   Column                                   Non-Null Count  Dtype 
---  ------                                   --------------  ----- 
 0   source sentence                          330 non-null    object
 1   simplified sentence                      330 non-null    object
 2   run_id                                   330 non-null    object
 3   snt_id                                   330 non-null    object
 4   No error                                 330 non-null    object
 5   Random generation                        330 non-null    bool  
 6   Syntax error                             330 non-null    bool  
 7   Contradiction                            330 non-null    bool  
 8   Simple punctuation / grammar errors      330 non-null    bool  
 9   Redundancy                               330 non-null    bool  
 10  Format misalignement                     330 non-null    bool 

In [5]:
import numpy as np
def nan_stats(data):
    total = data.isnull().sum()
    percent= np.round(data.isnull().sum()/data.isnull().count()*100, 2)
    nan_stats = pd.concat([total, percent],
                             axis=1, keys=['Total', '%']).sort_values('%', ascending=False)
    return nan_stats

In [6]:
#I'M USING THE FUNCTION THAT I DEFINED
print("MISSING VALUES PER COLUMNS , DESCENDING: ")
nan_stats(data)


MISSING VALUES PER COLUMNS , DESCENDING: 


Total     %
Commentaire                                297  90.0
simplified sentence                          0   0.0
Faithfulness hallucination                   0   0.0
Factuality hallucination                     0   0.0
Loss of Informative Content                  0   0.0
Overgeneralization                           0   0.0
Oversimplification of Logical Arguments      0   0.0
Topic shift                                  0   0.0
Out-of-Scope Generation                      0   0.0
Prompt misalignement                         0   0.0
source sentence                              0   0.0
Redundancy                                   0   0.0
Simple punctuation / grammar errors          0   0.0
Contradiction                                0   0.0
Syntax error                                 0   0.0
Random generation                            0   0.0
No error                                     0   0.0
snt_id                                       0   0.0
run_id                                       0   0.0
Format misalignement                         0   0.0

In [7]:
def data_describe(df):
    '''Check the number of rows, columns, missing values and duplicates.
       Count type of columns.
       Memory indication'''

    data_dict = {}
    data = df
    data_dict["merged dataframe"] = [data.shape[0],
                           data.shape[1],
                            round(data.isna().sum().sum()/data.size*100, 2),
                            round(data.duplicated().sum().sum()/data.size*100, 2),
                            data.select_dtypes(include=['object']).shape[1],
                            data.select_dtypes(include=['float']).shape[1],
                            data.select_dtypes(include=['int']).shape[1],
                            data.select_dtypes(include=['bool']).shape[1],
                            round(data.memory_usage().sum()/1024**2, 3)]

    comparative_table = pd.DataFrame.from_dict(data = data_dict,
                                                   columns = ['Rows', 'Columns', '%NaN', '%Duplicate',
                                                              'object_dtype','float_dtype', 'int_dtype',
                                                              'bool_dtype', 'MB_Memory'],
                                                   orient='index')
    print("SUMMARY FILES…")
    return(comparative_table)

In [8]:
data_describe(data)

SUMMARY FILES…


Rows  Columns  %NaN  %Duplicate  object_dtype  float_dtype  \
merged dataframe   330       20   4.5         0.0             6            0   

                  int_dtype  bool_dtype  MB_Memory  
merged dataframe          0          14       0.02

In [9]:
data.columns

Index(['source sentence', 'simplified sentence', 'run_id', 'snt_id',
       'No error', 'Random generation', 'Syntax error', 'Contradiction',
       'Simple punctuation / grammar errors', 'Redundancy',
       'Format misalignement', 'Prompt misalignement',
       'Out-of-Scope Generation', 'Topic shift',
       'Oversimplification of Logical Arguments', 'Overgeneralization',
       'Loss of Informative Content', 'Factuality hallucination',
       'Faithfulness hallucination', 'Commentaire'],
      dtype='object')

In [10]:
categorical_features = data.select_dtypes(include=['object', 'category']).columns.tolist()
print("categorical features :", categorical_features)

categorical features : ['source sentence', 'simplified sentence', 'run_id', 'snt_id', 'No error', 'Commentaire']


In [11]:

bool_columns = data.columns[data.dtypes == 'bool']

# Calculer les pourcentages de TRUE et FALSE pour chaque colonne booléenne
percentages = {
    col: {
        'TRUE (%)': data[col].mean() * 100,
        'FALSE (%)': (1 - data[col].mean()) * 100
    }
    for col in bool_columns
}

# Convertir en DataFrame pour afficher les résultats
result_df = pd.DataFrame(percentages).T

# Afficher les résultats
#import ace_tools as tools; tools.display_dataframe_to_user(name="Pourcentage des Colonnes Booléennes", dataframe=result_df)
print(result_df)


                                          TRUE (%)   FALSE (%)
Random generation                         4.545455   95.454545
Syntax error                             10.000000   90.000000
Contradiction                             0.000000  100.000000
Simple punctuation / grammar errors       9.090909   90.909091
Redundancy                                7.272727   92.727273
Format misalignement                      5.151515   94.848485
Prompt misalignement                      1.212121   98.787879
Out-of-Scope Generation                  13.939394   86.060606
Topic shift                               5.757576   94.242424
Oversimplification of Logical Arguments   7.575758   92.424242
Overgeneralization                       12.727273   87.272727
Loss of Informative Content              20.606061   79.393939
Factuality hallucination                  0.909091   99.090909
Faithfulness hallucination                8.181818   91.818182


En examinant la r´epartition des types d’erreurs dans le dataset, j’ai constat´e que la
classification serait d´es´equilibr´ee, car les pourcentages de la classe True sont tr`es faibles
pour la plupart des types d’erreurs. Certains types ne contiennent mˆeme aucune occur-
rence de True. Par cons´equent, j’ai d´ecid´e de limiter l’apprentissage `a quelques types
d’erreurs pour ce projet initial.
Pour commencer, j’ai choisi de me concentrer sur deux types d’erreurs sp´ecifiques :
• Loss of Informative Content : avec un pourcentage de True de 20.61 %.
• Out-of-Scope Generation : avec un pourcentage de True de 13.94 %.
Ces deux types pr´esentent les pourcentages de True les plus ´elev´es dans le dataset,
ce qui permet de les classifier de mani`ere plus r´ealiste. Cependant, avec un futur dataset
mieux ´equilibr´e et de plus grande taille, il sera envisageable d’´etendre cette approche `a
la classification de tous les types d’erreurs. Pour l’instant, ces choix permettent de poser
une base solide pour entamer le projet tout en tenant compte des limitations actuelles des
donn´ees

In [12]:
data.head()

source sentence  \
0   :[0],"non-coding RNAs (lncRNAs) may play crit...   
1  Recently, lncRNAs have been widely identified ...   
2  However, there is rare information about them ...   
3  During humanitarian crises, there is a need fo...   
4  Such need creates the base for misinformation ...   

                                 simplified sentence                   run_id  \
0   Long non-coding RNAs, or lncRNAs, may play si...  UBO_Task3.1_Phi4mini-sl   
1   Small RNA molecules are found in complex cell...  UBO_Task3.1_Phi4mini-sl   
2   There is unusual information here that doesn'...  UBO_Task3.1_Phi4mini-sl   
3  When crises happen, people need quick and larg...  UBO_Task3.1_Phi4mini-sl   
4  This reduction helps in combating climate chan...  UBO_Task3.1_Phi4mini-sl   

               snt_id No error  Random generation  Syntax error  \
0  G04.2_2519200480_1     VRAI              False         False   
1  G04.2_2519200480_2     VRAI              False         False   
2  G04.2_2519200480_3     FAUX              False         False   
3  G07.1_3007380484_1     FAUX              False         False   
4  G07.1_3007380484_2     FAUX              False         False   

   Contradiction  Simple punctuation / grammar errors  Redundancy  \
0          False                                False       False   
1          False                                False       False   
2          False                                False       False   
3          False                                False       False   
4          False                                False        True   

   Format misalignement  Prompt misalignement  Out-of-Scope Generation  \
0                 False                 False                    False   
1                 False                 False                    False   
2                 False                 False                     True   
3                 False                 False                     True   
4                 False                  True                    False   

   Topic shift  Oversimplification of Logical Arguments  Overgeneralization  \
0        False                                    False               False   
1        False                                    False               False   
2         True                                    False               False   
3        False                                    False               False   
4         True                                    False               False   

   Loss of Informative Content  Factuality hallucination  \
0                        False                     False   
1                        False                     False   
2                        False                     False   
3                        False                     False   
4                        False                     False   

   Faithfulness hallucination Commentaire  
0                       False         NaN  
1                       False         NaN  
2                       False         NaN  
3                       False         NaN  
4                       False         NaN

### PREPROCECCING

In [13]:
data.drop('run_id', axis=1, inplace=True)


In [14]:
data.drop('snt_id', axis=1, inplace=True)

In [15]:
data.head()

source sentence  \
0   :[0],"non-coding RNAs (lncRNAs) may play crit...   
1  Recently, lncRNAs have been widely identified ...   
2  However, there is rare information about them ...   
3  During humanitarian crises, there is a need fo...   
4  Such need creates the base for misinformation ...   

                                 simplified sentence No error  \
0   Long non-coding RNAs, or lncRNAs, may play si...     VRAI   
1   Small RNA molecules are found in complex cell...     VRAI   
2   There is unusual information here that doesn'...     FAUX   
3  When crises happen, people need quick and larg...     FAUX   
4  This reduction helps in combating climate chan...     FAUX   

   Random generation  Syntax error  Contradiction  \
0              False         False          False   
1              False         False          False   
2              False         False          False   
3              False         False          False   
4              False         False          False   

   Simple punctuation / grammar errors  Redundancy  Format misalignement  \
0                                False       False                 False   
1                                False       False                 False   
2                                False       False                 False   
3                                False       False                 False   
4                                False        True                 False   

   Prompt misalignement  Out-of-Scope Generation  Topic shift  \
0                 False                    False        False   
1                 False                    False        False   
2                 False                     True         True   
3                 False                     True        False   
4                  True                    False         True   

   Oversimplification of Logical Arguments  Overgeneralization  \
0                                    False               False   
1                                    False               False   
2                                    False               False   
3                                    False               False   
4                                    False               False   

   Loss of Informative Content  Factuality hallucination  \
0                        False                     False   
1                        False                     False   
2                        False                     False   
3                        False                     False   
4                        False                     False   

   Faithfulness hallucination Commentaire  
0                       False         NaN  
1                       False         NaN  
2                       False         NaN  
3                       False         NaN  
4                       False         NaN

### nettoyage des données textuelles

In [16]:
# import pandas as pd
# import re

# # Fonction de nettoyage du texte
# def clean_text(text):
#     # Supprimer le backslash avant les pourcentages et autres caractères spéciaux
#     text = re.sub(r'\\%', '%', text)  # Remplacer \% par %
    
#     # Supprimer les tirets multiples (---)
#     text = re.sub(r'---', ' ', text)
    
#     # Enlever la ponctuation non utile (comme ;, :, ., etc.)
#     text = re.sub(r'[^\w\s%.-]', '', text)
    
#     # Convertir le texte en minuscules, sauf pour les sigles/termes techniques
#     text = text.lower()
    
#     return text



# # Appliquer la fonction de nettoyage aux colonnes 'source sentence' et 'simplified sentence'
# data['source sentence'] = data['source sentence'].apply(clean_text)
# data['simplified sentence'] = data['simplified sentence'].apply(clean_text)




### NLP-SPACY PREPROCESSING

In [17]:
import spacy
nlp = spacy.load("en_core_web_lg")

In [18]:
type(nlp)

spacy.lang.en.English

In [19]:
nlp.lang

'en'

In [20]:
nlp.component_names

['tok2vec',
 'tagger',
 'parser',
 'senter',
 'attribute_ruler',
 'lemmatizer',
 'ner']

In [21]:
nlp.max_length

1000000

In [23]:
# import spacy
# from unidecode import unidecode  # Pour gérer les accents
# import pandas as pd

# # Charger le modèle SpaCy pour l'anglais
# nlp = spacy.load("en_core_web_lg")

# # Fonction pour nettoyer un texte avec tous les filtres
# def clean_text_with_spacy(text):
#     # Traiter le texte avec SpaCy
#     doc = nlp(text)
    
#     # Appliquer tous les filtres et transformations
#     cleaned_tokens = [
#         unidecode(token.lemma_.lower())  # Lemmatisation + conversion en minuscules + suppression des accents
#         for token in doc
#         if not token.is_punct           # Supprimer la ponctuation
#         and not token.is_space          # Supprimer les espaces
#         and not token.is_stop           # Supprimer les stopwords
#     ]
    
#     # Joindre les mots nettoyés pour reconstruire la phrase
#     return " ".join(cleaned_tokens)

# # Fonction pour nettoyer un corpus entier
# def clean_corpus(df, columns):
#     for col in columns:
#         df[f'cleaned_{col}'] = df[col].apply(lambda text: clean_text_with_spacy(str(text)))
#     return df


# df = pd.DataFrame(data)

# # Nettoyer les colonnes avec la fonction
# columns_to_clean = ['source sentence', 'simplified sentence']
# cleaned_df = clean_corpus(df, columns_to_clean)

# # Afficher le DataFrame nettoyé
# print(cleaned_df)
